### ISSUE TITLE
Secondary navigation item on hover does not meet contrast ratio requirements.

### ISSUE THREAD

**Issue by mhlozek** [Shopify/polaris-react#5925]
## Issue summary
Secondary navigation item on hover does not meet accessible contrast ratio requirements.

## Expected behavior
Minimal contrast ratio should be 4.5.

## Actual behavior
Text color: rgb(32,34,35)
Background color: rgb(109,113,117)

Contrast ratio: 4.39

## Steps to reproduce the problem
Hover the secondary navigation item on https://polaris.shopify.com/components/navigation/navigation#navigation

---

Shouldn't be used `var(--p-text);` color on hover/pressed state?
